test/auth.test.ts

```diff
@@ -0,0 +1,123 @@
+import {
+  getDefaultRoleForAllowedRoles,
+  getGroupsWithMappings,
+  getAllAllowedRolesForAuthGroups,
+  validateGroupRoleMappings,
+  mapGroupsToRoles
+} from '../src/packages/auth/functions';
+import { getEnv } from "../src/env";
+import { describe, expect, test, vi } from 'vitest'
+
+describe('auth helper function tests', () => {
+
+  test('mock env', () => {
+    const default_roles = ['role_1'];
+    vi.stubEnv("DEFAULT_ROLE", JSON.stringify(default_roles));
+
+    const { DEFAULT_ROLE } = getEnv();
+
+    expect(DEFAULT_ROLE[0]).toBe('role_1');
+  });
+
+  test('test default role priority', () => {
+    const default_roles = ['role_1', 'role_2'];
+    const allowed_roles = ['role_2', 'role_1'];
+
+    vi.stubEnv("DEFAULT_ROLE", JSON.stringify(default_roles));
+
+    expect(getDefaultRoleForAllowedRoles(allowed_roles)).toBe('role_1');
+  });
+
+  test(
+    'test auth group filter only considers groups that have a mapping', () => {
+
+      const auth_groups = ['ldap_example_admins', 'ldap_example_users'];
+      const group_role_mapping = {
+        "ldap_example_admins": ["role_1", "role_2"],
+        "ldap_example_viewers": ["role_3"]
+      };
+
+      vi.stubEnv("AUTH_GROUP_ROLE_MAPPINGS", JSON.stringify(group_role_mapping));
+
+      expect(getGroupsWithMappings(auth_groups)).toStrictEqual(["ldap_example_admins"]);
+
+  });
+
+  test(
+    'correct union of all allowed roles for auth groups', () => {
+
+      const auth_groups = ['ldap_example_admins', 'ldap_example_viewers'];
+      const group_role_mapping = {
+        "ldap_example_admins": ["role_1", "role_2"],
+        "ldap_example_viewers": ["role_2", "role_3"]
+      };
+
+      vi.stubEnv("AUTH_GROUP_ROLE_MAPPINGS", JSON.stringify(group_role_mapping));
+
+      expect(getAllAllowedRolesForAuthGroups(auth_groups)).toStrictEqual(["role_1", "role_2", "role_3"]);
+  });
+
+  test(
+    'group -> role mapping that doesnt contain default role throws error', () => {
+
+      const default_roles = ['role_1'];
+      const group_role_mapping = {
+        "ldap_example_admins": ["role_1", "role_2"],
+        "ldap_example_viewers": ["role_2", "role_3"]
+      };
+
+      vi.stubEnv("DEFAULT_ROLE", JSON.stringify(default_roles));
+      vi.stubEnv("AUTH_GROUP_ROLE_MAPPINGS", JSON.stringify(group_role_mapping));
+
+      expect(() => validateGroupRoleMappings())
+          .toThrowError('No roles within DEFAULT_ROLE list were found in the group to role mapping.');
+  });
+
+  describe(
+    'reference example', () => {
+
+      const default_roles = ['viewer', "user", "aerie_admin"];
```

Comment:
Minor, style: mixed quote styles in use here and across the file

Reply:
Good catch, I'll add a format check to CI as well.